File: Source/mvvm/src/main/java/cn/ung8023/MainActivity.java
package cn.ung8023;

import java.util.ArrayList;
import java.util.List;

import cn.ung8023.androidstudy.activity.ActivityEntity;
import cn.ung8023.androidstudy.activity.BaseMainActivity;
import cn.ung8023.mvvm.main.MVVMMainActivity;
import cn.ung8023.webview.WebViewMainActivity;

public class MainActivity extends BaseMainActivity {

    @Override
    protected List<ActivityEntity> getData() {
        return new ArrayList<ActivityEntity>(){
            {
                add(new ActivityEntity("MVVM", MVVMMainActivity.class));
                add(new ActivityEntity("WebView", WebViewMainActivity.class));
            }
        };
    }
}

File: Source/3rdframework/src/main/java/com/ung8023/a3rdframework/MainActivity.kt
package com.ung8023.a3rdframework

import cn.ung8023.androidstudy.activity.ActivityEntity
import cn.ung8023.androidstudy.activity.BaseMainActivity
import com.ung8023.a3rdframework.amazon.AmazonMainActivity
import com.ung8023.a3rdframework.dagger2.Dagger2MainActivity

class MainActivity : BaseMainActivity() {
    override fun getData(): MutableList<ActivityEntity> {
        return ArrayList<ActivityEntity>().apply {
            add(ActivityEntity("Amazon Cognito", AmazonMainActivity::class.java))
            add(ActivityEntity("Dagger2", Dagger2MainActivity::class.java))
        }
    }
}